src/controllers/authentication.ts

```diff
@@ -169,6 +174,89 @@ export const checkPassword = async (req: Request, res: Response) => {
   }
 };
 
+export const forgotPassword = async (req: Request, res: Response) => {
+  try {
+    const user = await userModel.User.findOne({ email: req.params.email });
+
+    if (!user) {
+      return res.status(200).json({
+        message: `Email has been sent to ${req.params.email}`,
+      });
+    } else {
+      const token = jwt.sign({ id: user._id }, jwtSecret, {
+        expiresIn: maxAge,
+      });
+
+      const port = "5173";
+      const link = `http://127.0.0.1:${port}/reset-password/${user._id}/${token}`;
```

Comment:
user id and token should be queryParams, if not it will be too hard to get those values in the frontend